### PR TITLE
feat: データとモデルの準備

### PR DESCRIPTION
**Pull request by maton369**

### CODE REVIEW

**Review by maton369 — COMMENTED** [review]
# 勾配の初期化（前エポックの値をクリア）
optimizer.zero_grad()
# 誤差逆伝播によって勾配を計算
loss.backward()
# 勾配に基づいてパラメータを更新
optimizer.step()
